Lib/test/test_xml_etree.py

```diff
@@ -122,6 +128,104 @@ def test_all(self):
         support.check__all__(self, ET, names, blacklist=("HTML_EMPTY",))
 
 
+class ElementTree_Element_UnitTest(unittest.TestCase):
+
+    def test___init__(self):
+        tag = "foo"
+        attrib = { "zix": "wyp" }
+
+        element_foo = ET.Element(tag, attrib)
+
+        with self.subTest("traits of an element"):
+            self.assertIsInstance(element_foo, ET.Element)
+            self.assertIn("tag", dir(element_foo))
+            self.assertIn("attrib", dir(element_foo))
+            self.assertIn("text", dir(element_foo))
+            self.assertIn("tail", dir(element_foo))
+
+        with self.subTest("string attributes have expected values"):
+            self.assertEqual(element_foo.tag, tag)
+            self.assertIsNone(element_foo.text)
+            self.assertIsNone(element_foo.tail)
+
+        with self.subTest("attrib is a copy"):
+            self.assertIsNot(element_foo.attrib, attrib)
+            self.assertEqual(element_foo.attrib, attrib)
+
+        with self.subTest("attrib isn't linked"):
+            attrib["bar"] = "baz"
+
+            self.assertIsNot(element_foo.attrib, attrib)
+            self.assertNotEqual(element_foo.attrib, attrib)
+
+
+    def test___copy__(self):
+        element_foo = ET.Element("foo", { "zix": "wyp" })
+        element_foo.append(ET.Element("bar", { "baz": "qix" }))
+
+        element_foo2 = copy.copy(element_foo)
+
+        with self.subTest("elements are not the same"):
+            self.assertIsNot(element_foo2, element_foo)
+
+        with self.subTest("string attributes are the same"):
+            self.assertIs(element_foo2.tag, element_foo.tag)
+            self.assertIs(element_foo2.text, element_foo.text)
+            self.assertIs(element_foo2.tail, element_foo.tail)
+
+        with self.subTest("string attributes are equal"):
+            self.assertEqual(element_foo2.tag, element_foo.tag)
+            self.assertEqual(element_foo2.text, element_foo.text)
+            self.assertEqual(element_foo2.tail, element_foo.tail)
+
+        with self.subTest("children are the same"):
+            for (child1, child2) in itertools.zip_longest(element_foo, element_foo2):
+                self.assertIs(child1, child2)
+
+        with self.subTest("attrib is a copy"):
+            self.assertEqual(element_foo2.attrib, element_foo.attrib)
+
+
+    def test___deepcopy__(self):
+        element_foo = ET.Element("foo", { "zix": "wyp" })
+        element_foo.append(ET.Element("bar", { "baz": "qix" }))
+
+        element_foo2 = copy.deepcopy(element_foo)
+
+        with self.subTest("elements are not the same"):
+            self.assertIsNot(element_foo2, element_foo)
+
+        # Strings should still be the same in a deep copy.
+        with self.subTest("string attributes are the same"):
```

Comment:
Equality is the correct check here, not identity. We are not testing deepcopy itself, only the deepcopying of Elements.

Reply:
I believe I was doing this for all `Element` attributes for completeness due to the discovery of the `attrib` inconsistency.